### ISSUE TITLE
🤔 might want to allow a plugin/optional dependency that can change the defaults

### ISSUE THREAD

**Issue by reactive-firewall** [reactive-firewall/multicast#31]
🤔 if there is a "default" should there be a way to change it?

_Originally posted by @reactive-firewall in https://github.com/reactive-firewall/multicast/pull/11#discussion_r813625348_

~- [ ] perhaps via a sysconf or something~ (perhaps in future)
~- [ ] or windows registry setting thinnggy (unlikely)~ (perhaps in future)
- [x] Overwrite defaults via env vars
- [x] and should document it

**Comment by reactive-firewall** [comment]
Note as for TTL values [RFC-1112](https://datatracker.ietf.org/doc/html/rfc1112#section-6) limits much of the value of such a feature. If these values are altered they should not be confusingly called 'defaults' perhaps take a hint from python and call them side-specific settings? 😵‍💫

**Comment by reactive-firewall** [comment]
@coderabbitai, please read-up on this ticket ( #31 ) and then read through all of the files in the repo focusing on any currently hardcoded values. Also read through the python files in the `multicast` directory focusing on default variables (including private ones). Finally compile a list of suggested variables to allow users to configure, (of course suggest a variable name if the hardcoded value does not have an appropriate variable to configure).

**Comment by reactive-firewall** [comment]
@coderabbitai, I'll implement them as a module named `env` in a file named `./multicast/env.py` ; while I implement the changes, can you please draft detailed specific patch notes including all of the key details you mentioned (of-course refactored for the `env` name instead of "conf")

**Comment by reactive-firewall** [comment]
I did a little refactoring to reduce the complexity of the load_config function (was triggering a C901 from flake8 tests with a value of 12 before)

in the spirit of single responsability I moved group, ttl, and port logic into their own respective functions:

```python
def load_port() -> int:
	"""
	Loads the value for _MCAST_DEFAULT_PORT from the environment var MULTICAST_PORT if able.
	"""
	# Import globals that we'll potentially update
	from multicast import _MCAST_DEFAULT_PORT
	try:
		port = int(os.getenv("MULTICAST_PORT", _MCAST_DEFAULT_PORT))
	except ValueError:
		warnings.warn(f"Invalid MULTICAST_PORT value, using default {_MCAST_DEFAULT_PORT}")
		port = _MCAST_DEFAULT_PORT
	# Validate and potentially update port
	if validate_port(port):
		globals()["""_MCAST_DEFAULT_PORT"""] = port
	else:
		warnings.warn(
			f"Port {port} is outside valid range (49152-65535), using default {_MCAST_DEFAULT_PORT}"
		)
		port = _MCAST_DEFAULT_PORT
	return port


def load_group() -> ipaddress.IPv4Address:
	"""
	Loads the value for _MCAST_DEFAULT_GROUP from the environment var MULTICAST_GROUP if able.
	"""
	# Import globals that we'll potentially update
	from multicast import _MCAST_DEFAULT_GROUP
	group = os.getenv("MULTICAST_GROUP", _MCAST_DEFAULT_GROUP)
	# Validate and potentially update group
	if validate_multicast_address(group):
		globals()["""_MCAST_DEFAULT_GROUP"""] = group
	else:
		warnings.warn(
			f"Invalid multicast group {group}, using default {_MCAST_DEFAULT_GROUP}"
		)
		group = _MCAST_DEFAULT_GROUP
	return ipaddress.IPv4Address(group)


def load_TTL() -> int:
	"""
	Loads the value for _MCAST_DEFAULT_TTL from the environment var MULTICAST_TTL if able.
	"""
	# Import globals that we'll potentially update
	from multicast import _MCAST_DEFAULT_TTL
	try:
		ttl = int(os.getenv("MULTICAST_TTL", _MCAST_DEFAULT_TTL))
	except ValueError:
		warnings.warn(f"Invalid MULTICAST_TTL value, using default {_MCAST_DEFAULT_TTL}")
		ttl = _MCAST_DEFAULT_TTL
	# Validate and potentially update TTL
	if validate_ttl(ttl):
		globals()["""_MCAST_DEFAULT_TTL"""] = ttl
	else:
		warnings.warn(
			f"TTL {ttl} is outside valid range (1-126), using default {_MCAST_DEFAULT_TTL}"
		)
		ttl = _MCAST_DEFAULT_TTL
	# Update socket default timeout
	socket.setdefaulttimeout(int(ttl))
	return ttl

```

please draft detailed [CEP-7]() aligned docstrings for each of these, be sure to include at least 3 Testcases for the "Minimum Acceptance Testing" doctests, and consider each line of code needs to be tested to maintain the project's test coverage.

**Comment by reactive-firewall** [comment]
@coderabbitai, also generate at-least 3 different, additional Testcases for expanding the doctests of the `load_config()` function that cover cases where "MULTICAST_BIND_ADDR" environment var is set (both valid and invalid cases), of course consider [CEP-7](https://gist.github.com/reactive-firewall/123b8a45f1bdeb064079e0524a29ec20) and testing BCP for any generated Testcases.

**Comment by reactive-firewall** [comment]
@coderabbitai, also generate at-least 3 different, additional Testcases for expanding the doctests of the load_config() function that cover cases where "MULTICAST_BUFFER_SIZE" environment var is set, of course consider [CEP-7](https://gist.github.com/reactive-firewall/123b8a45f1bdeb064079e0524a29ec20) and testing BCP for any generated Testcases.

**Comment by reactive-firewall** [comment]
@coderabbitai, please generate a specific markdown document for the new `multicast/env.py` module just implemented by this GHI ( #31 )
